codegen/src/generator/json.rs

```diff
@@ -97,6 +97,20 @@ impl GenerateProtocol for JsonGenerator {
         "f64"
     }
 
+    fn generate_additional_annotations(&self, service: &Service, shape_name: &str, type_name: &str) -> Vec<String> {
+        let service_name = service.service_type_name();
+
+        if (service_name == "DynamoDb" || service_name == "DynamoDbStreams") &&
+           (type_name == "ListAttributeValue" || type_name == "MapAttributeValue") {
+                vec![format!("#[serde(bound=\"\")]")]
+        } else if service.service_type_name() == "Emr" &&
```

Comment:
Can we reuse `service_name` here instead of calling `service_type_name()` again?